Fix: Gallery Block: Append images alt in aria-label attribute.

donmhico opened the pull request:
## Description
Fixes #16886.

Appends the images alt attributes (if any) in the aria-label.

## How has this been tested?
Followed the steps provided in #16886. 

## Screenshots
<img width="1039" alt="Screen Shot 2019-08-20 at 1 51 05 PM" src="https://user-images.githubusercontent.com/5747475/63320864-a3d62b00-c351-11e9-9752-d173722327f8.png">

## Types of changes
Fixes #16886

## Checklist:
- [x] My code is tested.

maxme committed:
Update video player style on mobile
- Add a new gridicon play icon, from: https://github.com/Automattic/gridicons/blob/87c9fce08b4a9f184b9fb4963228757fdd4f4e74/svg-min/gridicons-play.svg
- Replace the Dashicon play by this one
- Update icon size and icon color
- Update the overlay color

mchowning committed:
[RNMobile] Insure tapping at end of post inserts at end

Previously, tapping at the end of the post would insert a block
immediately after the currently selected block. In addition, this commit
is cleaning out a few unusued props in the block-list file.

ntwb committed:
Use `400` as a valid `font-weight`

donmhico committed:
Fix: Gallery Block: Append images alt in aria-label attribute.

Fixes #16886

jorgefilipecosta committed:
Fix: InnerBlocks template sync only on direct locked all (#16973)

getdave committed:
Check to ensure focus has intentionally left the wrapped component in `withFocusOutside` HOC (#17051)

* Adds check to determine whether event occured within element trapping focus

Previously there was no check to see whether the blur event occured within the wrapped component. If it occurs within then we do not want to trigger the `handleFocusOutside` callback because (by definition) the focus has not moved outside the wrapped element.

This may have been a regression introduced in https://github.com/WordPress/gutenberg/pull/16878

* Fix handling blur event when document is not actually focused

Previously we fixed the wrong problem. The actual issue is that previously we were handling document blur events as true “focus outside” events when in fact they should be ignored as if the document is not focused then presumably the user did not intent the blur.

* Update to test for document loss of focus scenario

jorgefilipecosta committed:
Fix: Required block appender styles are set on edit post (#16943)

jorgefilipecosta committed:
Document server functions in block style variations section (#16997)

* Expand the block style variations documentation to document server functions

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Update docs/designers-developers/developers/filters/block-filters.md

Co-Authored-By: Chris Van Patten <chris@vanpattenmedia.com>

* Fix line break and code block closing

mchowning committed:
[RNMobile] update mobile to not use ListEdit (#17070)

* [RNMobile] update mobile to not use ListEdit

This update is in response to changes made on the web side in #15113
that were causing a crash on mobile.

* Extract list block logic not shared with mobile to separate component

* Improve name of AdditionalSettings component

ellatrix committed:
Typewriter experience (#16460)

* Typewriter XP

* Address feedback

* Do not maintain scroll position when it's out of view

* Reset scroll position when aborting

* Set initial trigger %

* Add test for viewport

* Clean up

* Adjust doc

* Return children for IE

* Hide bottom space when there are metaboxes

* Debounce scroll with RAF

mchowning committed:
[RNMobile] Hide replaceable block when adding block (#16931)

jorgefilipecosta commented on packages/block-library/src/gallery/edit.js:
```diff
@@ -302,8 +302,10 @@ class GalleryEdit extends Component {
 					) }
 				>
 					{ images.map( ( img, index ) => {
+						const imgAlt = ( img.alt.length > 0 ) ? `${ img.alt }. ` : '';
```
We need to make img.alt is defined e.g: `img.alt && img.alt.length > 0`. Currently, if I upload an image to a gallery using drag & drop the editor crashes because the image does not contain an alt text.

donmhico replied:
Thanks @jorgefilipecosta. I've amended the commit to include this change. 

https://github.com/WordPress/gutenberg/blob/14b57c7ccdeee88cd04a68d3f7a2f3fa6c9e6a33/packages/block-library/src/gallery/edit.js#L304-L306

Sorry if I wasn't able to test the drag and drop. I'll be more careful the next time.

jorgefilipecosta reviewed (commented):
Hi @donmhico thank you for your contribution this change generally looks good 👍  I just found a bug that we need to address before merging.
Cc: @afercia in case there is some feedback for the aria-label text used in this PR.

jorgefilipecosta committed:
Add: Disabled block count in the block manager (#17103)

ellatrix committed:
Writing Flow/Quote: allow splitting (#17121)

* Writing Flow/Quote: allow splitting

* Add extra merge e2e test

gwwar committed:
Build: remove global install of latest npm since we want to use the paired node/npm version (#17134)

* Build: remove global install of latest npm since we want to use the paired node/npm version
* Also update travis to remove --latest-npm flag

noisysocks committed:
Project automation: Rewrite actions using JavaScript (#17080)

* Project automation: Rewrite actions using JavaScript

* Project automation: Don't transpile or install all dependencies

* Project automation: pull-request-automation -> project-management-automation

* Project automation: Add explanatory comment for `npm install` hack

* Project automation: Add debug statements

* Project automation: Don't use GitHub's debug() function

* Project automation: Use `payload` in tasks, not `context`

* Project automation: Link to the relevant GitHub documentation

ellatrix committed:
Writing Flow: allow undo of patterns with BACKSPACE and ESC (#14776)

* RichText/Patterns/Input Interaction: allow undo of patterns with BACKSPACE and ESC

* Only run input rules when inserting text

* Fix typo

* Simplify

youknowriad committed:
Apply box-sizing border-box properly to the notices components (#17066)

jorgefilipecosta committed:
Fix: Converting an image using an external source to a gallery does not work (#16548)

epiqueras committed:
Editor: Update the store to use Core Data entities. (#16932)

* Editor: Update the store to use Core Data entities.

* Editor: Fix selector test suites.

* Editor: Fix some legacy selectors and behaviors.

* Editor: Fix action tests.

* Editor: Fix remaining broken unit tests.

* Editor: Fix more tests.

* Editor: Fix more e2e test behaviors.

* Editor: Fix preview functionality.

* Core Data: Fix autosaves filtering.

* Editor: Don't make entity dirty with initial edits.

* Editor: Don't save if the post is not saveable.

* Core Data: Fix merged edits logic.

* Core Data: Fix undo to fit e2e expected behaviors.

* Core Data: Handle more change detection and saving flows.

* Block Editor: Fix undo level logic.

* Core Data: Clean up undo reducer comment.

* Editor: Make `serializeBlocks` a util.

* Core Data: Clarify raw attribute usage.

* Core Data: Memoize .

* Core Data: Use new raw entity record selector instead of modifying the existing one.

* Core Data: Make save notices the caller's responsibility.

* Editor: Use the store key constant in actions instead of a string literal.

* Editor: Defer serialization of blocks until save.

* Editor: Fix raw content access in set up.

* Editor: Revert broken test change.

* Editor: Make initial edits a dirtying operation.

* Editor: Add comment clarifying why we set content to a new function on edits.

* Demo: Fix tests to consider the initial edits dirtying.

* Core Data: Set auto-drafts to drafts when autosaving them.

* Core Data: Handle receiving autosaves correctly when editing non-autosave-persisting-properties.

mikejolley committed:
Add callbacks to ServerSideRenderer to handle failures with custom renderers (#16512)

* Add callbacks to ServerSideRenderer to handle failures

* Update variable naming and set default renderers

* LoadingResponsePlaceholder

* swtich case

* fetchComplete

* Remove fetchcomplete method

* Pass render props

* add changelog entry

* add documentation for new props

noisysocks committed:
Project management automation: Fix 'add first time contributor label' (#17156)

GitHub returns the search result count as `data.total_count`, not
`total_count`.

gziolo committed:
Scripts: Improve recommended settings included in the package (#17027)

* Babel Preset Default: Add missing @wordpress/element dependency

* Remove `@wordpress/dependency-group` and `@wordpress/gutenberg-phase` rules from the `custom` and `recommended` configs and leave them as opt-in features.

* ESLint Plugin: Extract 2 test configs and add them conditionally to the recommended one

* Add missing documentation for changes applied

* Scripts: Update CHANGELOG file

noisysocks committed:
Project management automation: Fix 'add milestone' (#17157)

Octokit returns all resources wrapped in a `data` key, e.g.
`data.milestone` instead of `milestone`. Also,
`octokit.repos.getContents()` returns a base64 encoded string.

jorgefilipecosta committed:
chore: Remove unused css from ColorPalette component (#17152)

youknowriad committed:
Add the block example API and use it for inserter and switcher previews (#17124)

jorgefilipecosta committed:
Fix: Block toolbar appears above sidebar on medium viewports (#17108)

bfintal committed:
eslint BaseControl fix for TypeError: Cannot read property 'name' of undefined (#17044)

Fixes #17043

jrchamp committed:
Docs: Fix creating-dynamic-blocks typo (#17172)

senadir committed:
remove selection toggle on image (#17175)

Commit:
Add circle-crop variation to Image block. (#16475)

* Add circle-crop variation to Image block.

Props @phpbits for the technical implementation.

This PR adds a variation to the Image block, "Circle Crop". What that means is you can choose a style variation for the image, which rounds all 4 corners. When this is appliedto a square image, that means a perfect circle. When it is applied to a rectangle, it means a pill-shape.

In past attempts to fix this, it was either an oval crop, or it required a large refactor to be able to make a rectangular image into a square, in CSS only. But with @phpbits clever trick — using a pixel value for the border radius — the we can actually implement this in a simple way that is still visually appealing even for rectangles.

* If the browser supports it, use the mask CSS property to create this effect.

* Rename to "Circle Mask"

* Remove unintentional code comment.

jorgefilipecosta committed:
Add: Option to select the style that is automatically applied (#16465)

Squashed commits:
[ffd16b5f3] Add unit tests
[51dd9dc8b] Refactor & changed the design
[3b9ac24dd] Add: Option to allow user to select the style that is automatically applied.

adamsilverstein committed:
Enable an optional namespace parameter for hasAction & hasFilter (#15362)

* from https://github.com/WordPress/packages/pull/106/files

* Update packages/hooks/src/createHasHook.js

Co-Authored-By: Pascal Birchler <pascal.birchler@gmail.com>

* clean up docblock spacing

* add changelog entry for new namespace parameter for hasAction & hasFilter

* Update packages/hooks/CHANGELOG.md

Co-Authored-By: Grzegorz (Greg) Ziółkowski <grzegorz@gziolo.pl>

* break out tests

* update readme

* Apply suggestions from code review

Co-Authored-By: Daniel Richards <daniel.richards@automattic.com>

* Update CHANGELOG.md

jorgefilipecosta committed:
Add get_item_schema function to WP_REST_Widget_Areas_Controller (#15981)

ntwb committed:
fix(scripts): Use the SCSS shared stylelint-config-wordpress config (#17060)

* fix(scripts): use the SCSS shared stylelint-config-wordpress config

* Update CHANGELOG.md

senadir committed:
Add option to resize Cover Block (#17143)

* add resize support to cover block

* add height settings to option

* fix issue with setting min height

* fix deleted file

* change from height to minHeight

* refactor how changes are handled

* fix problem with resizing to smaller sizes

donmhico committed:
Fix: Empty Classic Editor inside innerBlock fatal error (#17164)

* Fix: Empty Classic Editor insider innerBlock fatal error

* Expand inline comment

senadir committed:
Allow directly setting a background to Cover (#17041)

* extend color palette & add docs

* extend placeholder to accept append prop

* extend media placeholder to pass down append prop to placeholder

* add palceholder margin

* add color support to cover placeholder

* add max width to colors

* fix typo

* add clear button

* fix panel settings ordering

* add reset button to cover

* add filename to settings panel

* fix review issues

* Update text on clear button.

* fix double color palette

* rebase & merge master

* remove append in favor of children

* Update docs

* Remove name attribute.

jorgefilipecosta committed:
Bump plugin version to 6.4.0-rc.1

afercia commented:
@donmhico @jorgefilipecosta thanks for working on this!.

> in case there is some feedback for the aria-label text used in this PR 

I'd say that if we want to keep the `n of n` information, it's fine to use the aria-label. However, it'd be great to adjust a couple things.

**When the alt text is empty:**
For consistency, I'd recommend to use the same fallback mechanism used for the image, see https://github.com/WordPress/gutenberg/blob/1383a598a2ac53ebb9d50fe1ccfa067b1d06c02a/packages/block-library/src/image/edit.js#L794-L802

It works thanks to a small utility function `getFilename()` (which in turn uses `getPath` imported from `@wordpress/url`). This mechanism allows screen reader users to have some context on the image even when the alt attribute is empty. Maybe worth considering to abstract `getFilename()`? @jorgefilipecosta I'd defer to you about this.

**Simplify the string**
Screen readers already announce the element type. There's no need to repeat "image". I know this is pre-existing but it's a good opportunity to remove the word "image", otherwise it will get repeated:

<img width="768" alt="Screenshot 2019-08-26 at 19 13 48" src="https://user-images.githubusercontent.com/1682452/63712121-4f451b00-c83d-11e9-8ca6-93dc8e3d1113.png">

I'd also remove "in gallery", as users already know they're editing a gallery. So the string could be something like:

`%1$s%2$d of %3$d`

This would be announced by screen readers as:

`Unicorns! 2 of 3`

or (with the fallbacks):

`This image has an empty alt attribute; its file name is unicorn-wallpaper-1024x768.jpg 2 of 3`

or (last fallback):

`This image has an empty alt attribute 2 of 3`

Note: the translators comment should be updated, as there are now 3 placeholders.

brentswisher committed:
Update the post schedule label to correctly reflect the date and (#15757)

time display settings

donmhico committed:
Use the image alt logic in image block to determine the aria-labels in gallery block.

donmhico commented:
Thanks for the input @afercia. Can you look in my latest commit https://github.com/WordPress/gutenberg/pull/17102/commits/503a80493f32418715c617f3ea9287f3c56b92be? It address the points you mentioned.

Also I think I messed up this PR when I tried to rebase this PR with the latest master. Let me know if I need to create a new PR with a new branch.

Thank you.

jorgefilipecosta commented:
Hi @donmhico, thank you for the updates 👍  It seems there we a problem during the rebase of this PR and it now has lots of conflicts and includes unrelated changes.

guarani commented:
Looks like this needs to be rebased with `master` and conflicts resolved to be testable on mobile since this PR pre-dates the monorepo changes.

guarani commented:
fyi GitHub has a limit of 15 reviewers so when I remove some mobile reviewers others get removed too (I suppose at some point in the past GitHub allowed more than 15 reviewers to be added).

glendaviesnz commented:
With the merging of the [gallery block refactor](https://github.com/WordPress/gutenberg/pull/25940) the gallery images are now Image blocks, which don't currently have an aria label, so alt tag should be read by default.
